runtime-common/stdlib/msgpack/zone.h

```diff
@@ -27,12 +30,14 @@ class zone : private vk::not_copyable {
     size_t m_free{0};
     char* m_ptr{nullptr};
     chunk* m_head{nullptr};
+    bool m_error{false};
   };
 
   char* allocate_expand(size_t size);
 
   size_t m_chunk_size{0};
   chunk_list m_chunk_list;
+  bool m_error{false};
```

Comment:
It seems that `zone` only has error in case of bad alloc. Isn't it enough to just check for `nullptr`?

Reply:
I think it will. Now you have two sources of truth: `m_error` and the actual result of `alloc_align` for example

Reply:
fixed